### ISSUE TITLE
[Android] Xenoblade 3 shader compilation error results in game crash and unusable shader cache

### ISSUE THREAD

**Issue by NocturnalBlue** [yuzu-emu/yuzu#11592]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

Early Access 2921a2426

### Description of Issue

While playing through Chapter 1 of the game, I encountered a game crash that results in being returned to Yuzu's game selection menu. It occurs during the cutscene showing Noah and Mio's first interlink, more specifically in the memory sequence, just after a white screen transition. After the crash, trying to enter the game again fails during the "Building Shaders" message. I've found that deleting the game's shader cache is the only way to re-enter the game.

After encountering this issue, I tried using the newer turnip 23.3.0 revision 6 driver from two weeks ago, however that driver results in similar crashes just by opening the pause menu. Again, rendering the shader cache unusable.

### Expected Behavior

No crashes.

### Reproduction Steps

Enter the first Moebius fight in chapter 1 and let the subsequent cutscene play.

### Log File

[yuzu_log_XC3.txt](https://github.com/yuzu-emu/yuzu/files/12719626/yuzu_log_XC3.txt)

### System Configuration

CPU: Snapdragon 8 gen 2
GPU/Driver: Adreno 740 - Turnip-23.3.0-vulkan-1.3.255
RAM: 12 GB
OS: Android 13

**Comment by NocturnalBlue** [comment]
As of today, October the 13th, the crashes that I have encountered have been resolved with the release of turnip-2.3.0-A7XX-rev7. This confirms my suspicions that it was indeed a driver issue. In my opinion this unfortunately means that it becomes difficult to correctly identify wether a graphics bug is inherent to yuzu android builds, or if it's caused by drivers. Once again, this drives home how bad the situation is with android drivers.
